src/go/serviceconfig/service_config_fetcher.go

```diff
@@ -19,39 +19,41 @@ import (
 	"crypto/x509"
 	"fmt"
 	"io/ioutil"
-	"math"
 	"net/http"
 	"time"
 
-	"github.com/GoogleCloudPlatform/esp-v2/src/go/metadata"
 	"github.com/GoogleCloudPlatform/esp-v2/src/go/options"
 	"github.com/GoogleCloudPlatform/esp-v2/src/go/util"
 	"github.com/golang/glog"
 	"github.com/golang/protobuf/proto"
 
 	confpb "google.golang.org/genproto/googleapis/api/serviceconfig"
-	smpb "google.golang.org/genproto/googleapis/api/servicemanagement/v1"
 )
 
+type GetAccessTokenFunc func() (string, time.Duration, error)
+type GetNewConfigIdFunc func() (string, error)
+
 type ServiceConfigFetcher struct {
 	serviceName         string
 	checkRolloutsTicker *time.Ticker
 	client              http.Client
-	mf                  *metadata.MetadataFetcher
-	opts                options.ConfigGeneratorOptions
+	opts                *options.ConfigGeneratorOptions
+
+	accessToken GetAccessTokenFunc
+	newConfigId GetNewConfigIdFunc
 
 	curServiceConfig *confpb.Service
-	curRolloutId     string
 }
 
-func NewServiceConfigFetcher(mf *metadata.MetadataFetcher, opts options.ConfigGeneratorOptions, serviceName string) (*ServiceConfigFetcher, error) {
+func NewServiceConfigFetcher(opts *options.ConfigGeneratorOptions,
+	serviceName string, accessTokenFromImds func() (string, time.Duration, error)) (*ServiceConfigFetcher, error) {
```

Comment:
The reason I asked to type alias the functions was so you can use the type aliases here as well to make it clearer.

```
func NewServiceConfigFetcher(opts *options.ConfigGeneratorOptions,
	serviceName string, accessTokenFromImds GetAccessTokenFunc) (*ServiceConfigFetcher, error) {
```

See if they can be used elsewhere as well. It makes it a lot easier to understand.